packages/app/src/webapp/views/sidecar.ts

```diff
@@ -389,7 +389,7 @@ export const setMaximization = (tab: Tab, op = 'add', cause: MaximizationCause =
   }
 
   tab.classList[op]('sidecar-full-screen')
-  setTimeout(() => eventBus.emit('/sidecar/maximize'), 600)
+  setTimeout(() => eventBus.emit('/sidecar/maximize'), 1000)
```

Comment:
the sidecar sweeps out with a transition duration of 600ms. the goal is to notify views that the sidecar is open and ready, so that they can do any "right sizing". the monaco-editor is the main one, because it (nastily!) uses fixed pixel sizes, rather than relative sizings.

600ms has been fine till now. but in firefox, there seems to be some extra delays

Reply:
Couldn't we implement the sidecar to be able to launch an event when the transition is complete?Having a `setTimeout` doesn't look like the right decision because it depends on the browser and the computer where it is executed.

Reply:
we are using a css transition. the underlying problem here is that monaco-editor isn't designed for this kind of thing; in its use of fixed positioning (pretty much everywhere), it cannot adopt to size transitions.

even if we used a javascript-governed transition, we cannot know when the browser's rendering engine will actually be done. the only two choices are to stop using monaco, or to stop using width transitions.

Reply:
If we are using CSS transitions, can we use the `transitionend` event?

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/transitionend_event